### PR TITLE
Allow again some seed taints to be used

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/kind cleanup
/priority normal

**What this PR does / why we need it**:
Allows again some seed taints to be used.

**Which issue(s) this PR fixes**:
Follow up on #2955 

**Special notes for your reviewer**:
Merge after v1.12 is released.

**Release note**:

```improvement operator
It is again allowed the seed taints `seed.gardener.cloud/disable-capacity-reservation`, `seed.gardener.cloud/disable-dns` and `seed.gardener.cloud/invisible` to be used. Note, these taints have been replaced by `seed.spec.settings` fields and there is no special semantic behind them anymore.
```

### CODE REVIEW

**Comment by timuthy** [comment]
Hm I was think if it makes sense to remove them only in `v1.12`, so only for a short period of time. Maybe it's already fine to keep them in general but announce in the release notes of `v1.12` that these taints are not synced any more. WDYT?

**Comment by vpnachev** [comment]
Initially I had no strong opinion about keeping/removing them, but I think it is nice to remove them automatically in a single minor version to avoid manual operations. If they were left, then shoots without the respective toleration cannot be scheduled on the seed.

**Comment by timuthy** [comment]
> If they were left, then shoots without the respective toleration cannot be scheduled on the seed.

Wouldn't that be the case already?

**Comment by vpnachev** [comment]
Usually yes, but I am targeting the case where the `spec.settings.*` field  is changed and then the respective taint is not removed.

**Comment by rfranzke** [comment]
Now that v1.12.0 got released I guess this PR is
/ready

File: pkg/apis/core/v1alpha1/defaults_test.go [hunk 1]
```diff
@@ -233,38 +233,23 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Settings.VerticalPodAutoscaler.Enabled).To(BeTrue())
 		})
 
-		It("should remove deprecated taints from the seed settings (w/ taints)", func() {
-			obj.Spec.Taints = []SeedTaint{
+		It("should allow taints that were not allowed in verison v1.12", func() {
```
Comment (rfranzke) [inline pkg/apis/core/v1alpha1/defaults_test.go]:
```suggestion
		It("should allow taints that were not allowed in version v1.12", func() {
```

File: pkg/apis/core/v1beta1/defaults_test.go [hunk 2]
```diff
@@ -234,38 +234,23 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Settings.VerticalPodAutoscaler.Enabled).To(BeTrue())
 		})
 
-		It("should remove deprecated taints from the seed settings (w/ taints)", func() {
-			obj.Spec.Taints = []SeedTaint{
+		It("should allow taints that were not allowed in verison v1.12", func() {
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/defaults_test.go]:
```suggestion
		It("should allow taints that were not allowed in version v1.12", func() {
```

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/apis/core/v1alpha1/defaults_test.go (hunk 1 above)
Comment (vpnachev) [reply pkg/apis/core/v1alpha1/defaults_test.go]: Thanks!

File: pkg/apis/core/v1alpha1/defaults_test.go [hunk 3]
```diff
@@ -233,38 +233,23 @@ var _ = Describe("Defaults", func() {
...
-			obj.Spec.Taints = []SeedTaint{
+		It("should allow taints that were not allowed in version v1.12", func() {
+			taints := []SeedTaint{
 				{Key: "seed.gardener.cloud/disable-capacity-reservation"},
 				{Key: "seed.gardener.cloud/disable-dns"},
 				{Key: "seed.gardener.cloud/invisible"},
 			}
+			obj.Spec.Taints = taints
 
+			obj.DeepCopy()
```
Comment (ialidzhikov) [inline pkg/apis/core/v1alpha1/defaults_test.go]:
```suggestion
```

File: pkg/apis/core/v1beta1/defaults_test.go [hunk 4]
```diff
@@ -234,38 +234,23 @@ var _ = Describe("Defaults", func() {
...
-			obj.Spec.Taints = []SeedTaint{
+		It("should allow taints that were not allowed in version v1.12", func() {
+			taints := []SeedTaint{
 				{Key: "seed.gardener.cloud/disable-capacity-reservation"},
 				{Key: "seed.gardener.cloud/disable-dns"},
 				{Key: "seed.gardener.cloud/invisible"},
 			}
+			obj.Spec.Taints = taints
 
+			obj.DeepCopy()
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/defaults_test.go]:
```suggestion
```

**Review by rfranzke — APPROVED** [review]

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**